app/models/course/assessment/submission.rb

```diff
@@ -86,6 +86,38 @@ class Course::Assessment::Submission < ActiveRecord::Base
   #   Orders the submissions by date of creation. This defaults to reverse chronological order
   #   (newest submission first).
   scope :ordered_by_date, ->(direction = :desc) { order(created_at: direction) }
+
+  # @!method self.ordered_by_submitted date
+  #   Orders the submissions by date of submission (newest submission first).
+  scope :ordered_by_submitted_date, (lambda do
+    all.calculated(:submitted_at).order('submitted_at DESC')
+  end)
+
+  # @!method self.ordered_by_workflow_state
+  #   Orders the submissions by workflow_state. This defaults to reverse chronological order.
+  scope :ordered_by_workflow_state, ->(direction = :desc) { order(workflow_state: direction) }
+
+  # @!method self.confirmed
+  #   Returns submissions which have been submitted (which may or may not be graded).
+  scope :confirmed, -> { where(workflow_state: [:submitted, :graded]) }
+
+  # @!method self.for_course_staff
+  #   Returns all confirmed submissions (status: submitted or graded), ordered firstly by the
```

Comment:
@weiqingtoh I think we should not document "how" and "what" in comments, that should be the responsibility of the code (i.e. your code should be expressive and intention-revealing). Document this way will increase difficulty of maintenance IMHO: every time we update the code we will need to update the document. And the comments will become harmful if we forgot to update them.

Instead we should document "why", why you wrote this piece of code and for what.

What do you think ?